### PR TITLE
doc: Clarify take kernel semantics

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-select/src/take.rs [hunk 1]
```diff
@@ -50,6 +50,12 @@ use num::{One, Zero};
 ///
 /// For selecting values by index from multiple arrays see [`crate::interleave`]
 ///
+/// Note that the semantics of this kernel ("take" elements) is reflected on the output
+/// values. The corresponding elements in the output array are the elements in the input
+/// array at the indices specified in the indices array. This kernel, similar to other
+/// kernels in this crate, does not guarantee the physical memory layout of the output.
```
Comment (Dandandan) [inline arrow-select/src/take.rs:56]: I think this phrasing might be a bit confusing as the physical memory layout (i.e. arrow arrays) is still guaranteed.

What isn't guaranteed is whether a fresh copy is created or buffers are reused from existing array(s).

Comment (viirya) [reply arrow-select/src/take.rs:56]: Okay

File: arrow-select/src/take.rs [hunk 2]
```diff
@@ -50,6 +50,12 @@ use num::{One, Zero};
 ///
 /// For selecting values by index from multiple arrays see [`crate::interleave`]
 ///
+/// Note that the semantics of this kernel ("take" elements) is reflected on the output
+/// values. The corresponding elements in the output array are the elements in the input
+/// array at the indices specified in the indices array. This kernel, similar to other
+/// kernels in this crate, does not guarantee that it creates new buffers or takes/reuses
+/// existing buffers from the input arrays.
```
Comment (tustvold) [inline arrow-select/src/take.rs:57]:
```suggestion
/// Note that this kernel, similar to other kernels in this crate,
/// will avoid allocating where not necessary. Consequently
/// the returned array may share buffers with the inputs
```

Perhaps?

I at least found the section about semantics being reflected on output values somewhat hard to interpret

Comment (viirya) [reply arrow-select/src/take.rs]: Okay

**Comment by viirya** [comment]
Thanks @Dandandan @tustvold

**Pull request merged**